### ISSUE TITLE
Check versioning

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/pelican-frontend#116]
Pelican backend sets versions on (most) checks. The idea was that the JSON blob can be modified, and the version changed – allowing Pelican backend to change freely, while giving other components an opportunity to handle the change.

Until ae0b07d, the only code in Pelican frontend that compared versions was in datasetMixins.js, and its only effect was to not return a dataset-level check's type if the version didn't match (resulting in the check not being rendered).

For now, I've removed that code, and our approach going forward can be:

1. Make changes to Pelican backend conservatively.
2. Immediately make changes to Pelican frontend, when necessary.
3. Add backwards-compatibility in Pelican frontend as needed, so that it can render older reports.

In other words, we're using open ended requirements from Pelican frontend's perspective, rather than `<=version` logic, which is more conservative about the likelihood of breaking changes from a dependency.

**Comment by jpmckinney** [comment]
This policy is now documented in the docs.
